packages/studio-be/src/core/misc/archive.ts

```diff
@@ -1,6 +1,60 @@
+import fse from 'fs-extra'
+import glob from 'glob'
+import mkdirp from 'mkdirp'
+import path from 'path'
+import stream from 'stream'
 import tar from 'tar'
+import tmp from 'tmp'
+import unzipper from 'unzipper'
 import { VError } from 'verror'
 
+import { forceForwardSlashes } from './utils'
+
+// Source: https://github.com/kevva/is-zip
+const isZip = buf => {
+  if (!buf || buf.length < 4) {
+    return false
+  }
+
+  return (
+    buf[0] === 0x50 &&
+    buf[1] === 0x4b &&
+    (buf[2] === 0x03 || buf[2] === 0x05 || buf[2] === 0x07) &&
+    (buf[3] === 0x04 || buf[3] === 0x06 || buf[3] === 0x08)
+  )
+}
+
+export const extractArchive = async (archive: Buffer, destination: string): Promise<string[]> => {
+  try {
+    if (!(await fse.pathExists(destination))) {
+      await mkdirp(destination)
+    }
+
+    const buffStream = new stream.PassThrough()
+    buffStream.end(archive)
+
+    let writeStream
+    if (isZip(archive)) {
+      writeStream = unzipper.Extract({ path: destination })
+    } else {
+      writeStream = tar.extract({ strict: true, cwd: destination })
+    }
+
+    buffStream.pipe(writeStream)
+
+    await new Promise((resolve, reject) => {
+      writeStream.on('close', resolve) // emitted by unzipper
+      writeStream.on('end', resolve) // emitted by tar
+      writeStream.on('error', reject)
+    })
+
+    const files = await Promise.fromCallback<string[]>(cb => glob('**/*.*', { cwd: destination }, cb))
+    return files.map(filePath => forceForwardSlashes(filePath))
+  } catch (err) {
+    throw new VError(err, `[Archive] Error extracting archive to "${destination}"`)
```

Comment:
There seem to have a typings issue with the err:

![Screenshot from 2021-09-09 17-22-38](https://user-images.githubusercontent.com/9640576/132764573-2e1810be-3926-479d-abb1-b375418fb0c7.png)

Reply:
This is related to a new typescript version in vscode, unrelated